core/lib/tracehouse/trace-processor.js

```diff
@@ -488,21 +488,23 @@ class TraceProcessor {
    * @return {Map<number, number>} Map where keys are process IDs and their values are thread IDs
    */
   static findMainFramePidTids(mainFrameInfo, keyEvents) {
-    const frameCommittedEvts = keyEvents.filter(evt =>
-      evt.name === 'FrameCommittedInBrowser' &&
+    const frameProcessEvts = keyEvents.filter(evt =>
+      // ProcessReadyInBrowser is used when a processID isn't available when the FrameCommittedInBrowser trace event is emitted.
+      // In that case. FrameCommittedInBrowser has no processId, but a processPseudoId. and the ReadyInBrowser event declares the proper processId.
```

Comment:
total nit, but maybe move the `filter(Boolean)` below into this filter (as `&& e?.args?.data?.processId` or whatever) so it's closer to the comment explaining it

Reply:
I guess it's less clear then that any `undefined` from `e?.args?.data?.processId` is immediately filtered out (instead it's pre-filtered), though tsc doesn't care (`mainFramePids` is `Array<number|undefined>` either way)

Reply:
its a good idea. i did it.